execution/executor/src/lib.rs

```diff
@@ -735,7 +736,7 @@ impl<V: VMExecutor> BlockExecutor for Executor<V> {
             });
         }
 
-        if version + 1 == num_persistent_txns {
+        if version.saturating_add(1) == num_persistent_txns {
```

Comment:
As you're in a function that return a result, it is better to use `checked_add` rather than `saturating_add`, and return an error if `checked_add` returns `None`. I don't think there's a risk in using `saturating_add` here, but we can never be too sure.